### ISSUE TITLE
The settings.json will be over wrote after the transmission-daemon start

### ISSUE THREAD

**Issue by chenyz1984** [transmission/transmission#882]
I install the transmission through the source code in the debian 9, and  set the --prefix to /var/local/transmission. I modify the default settings.json such as the download-dir to '/mnt/hgst/download', But after start the transmission-daemon by command 'transmission-daemon -f --log-error -g /var/local/transmission/etc'. the parameter (download-dir) will be set to '/root/Downloads'.

**Comment by roobandev** [comment]
@chenyz1984  You need to stop  daemon before editing settings.json. Start it once you are done editing.

**Comment by AngelFalse** [comment]
Are you sure that, while you're editing the config file, the transmission-daemon is stopped?

Because I always edit config files when the daemon is stopped, and it never rewrite the file unconditionally while being started.

**Comment by tq02ksu** [comment]
Why transmission-daemon write back the settings file before stop, many people boring with this problem and discussing at many sites, can anybody give me some tips about the design, take me to the insight.

**Comment by razum2um** [comment]
Generally I agree about unusual behaviour, because unix-way taught us to "edit config and restart", not "stop, edit config and start", I really don't remember any other OSS behaving this way.

@tq02ksu the reason is ability to save values which can be changed in web-ui, but still, not sure, why it's not rewritten immediately, maybe due performance reasons.. another reason I see - to ensure config is valid, but this is even more strange.

imo it's worth another "readonly-config" options

Besides, be aware that this behaviour is not valid if launched inside k8s or similar envs, where ConfigMap is mounted read-only. more: https://twitter.com/razum2um/status/1206989790643675136

TLDR: I suggest to handle

**Comment by thcipriani** [comment]
> Generally I agree about unusual behaviour, because unix-way taught us to "edit config and restart", not "stop, edit config and start", I really don't remember any other OSS behaving this way.

This is very confusing and frustrating behavior. Overwriting user configuration in some circumstances (exact circumstances remain opaque to me) is not following the [principle of least astonishment](https://en.wikipedia.org/wiki/Principle_of_least_astonishment) for software design.

(CC @sandrotosi)

**Comment by Scottpedia** [comment]
I am sorry but I have to express my frustration against this stupid design here. The fact that there are even multiple config file scattered across the OS filesystem leads many of us into thinking about that the problem is about the right one to use. And there is no mention at all about this mechanism on the man page, what a joke? I urge the dev team to handle this issue ASAP, and investigate the reason why it exists.

@ckerr

**Comment by Scottpedia** [comment]
Hey can you also change the security measures put in place to "protect" the users who in most cases know what they are doing? wdym by:
```
To fix this, choose one of the following options:

    Enable password authentication, then any hostname is allowed.
    Add the hostname you want to use to the whitelist in settings.
```
I just want to say it loud that : **WE KNOW WHAT WE ARE DOING.** By adding this measure, it feels like the program is acting against its users which is not compatible with the ideals of the free software movement! 

WE DON'T NEED YOUR "PROTECTION"!

**Comment by OliverLew** [comment]
My idea would be, update the settings.josn every time user changes some settings through GUI/WebUI/other client. And do not update when quiting transmission in case user wants to do "edit and restart".

**Comment by dmak** [comment]
One can make things easier if there is a way to tell Transmission daemon to flush configuration:
* By sending it a signal, e.g. `SIGUSR1`
* By making an RPC call: `transmission-remote --save-config`
* By introducing an option in the daemon to set autosave interval: `transmission-daemon --config-autosave-interval=1m`

Nobody is protected from unexpected segfaults in Transmission code or hardware interruptions – I believe the changes made in GUI are lost in this case?

**Comment by quite** [comment]
At least there it is possible to have the daemon reload the `settings.json` by sending it a `SIGHUP`. This seems to have been possible since around 2010. So you can edit the json, and have transmission-daemon reload it. But sure, having a way to trigger transmission-daemon writing down the json would be useful, for when you cause setting changes in another way, and want them on disk (to edit them later for example).

**Comment by progserega** [comment]
I can not change password to rpc login:

1. setup transmission-daemon - it generate password such: "{58d6b6b4691a50aa67edee69de75ddec9324168d9D3GEo7W"
2. I try connect over "Transmission remote GUI" app - it sayed, that "passord DO NOT contain symbols '{}'"
3. I stop transmission daemon
4. change one symbol in password from { to _: "_58d6b6b4691a50aa67edee69de75ddec9324168d9D3GEo7W"
5. start transmission-daemon
6. show config and get NEW password with {: "{8a23e376a2ceb2e6cf46d1324fee5c3949e3f226Z3dW/qi/"

This behavior is very unnerving.

debian 13:
transmission-daemon           4.1.0~beta2+dfsg-3+deb13u1
